File: collect_odd_values.rb
def collect_odd_values(arr)
  result = []

  def odd_collector(collection, result)
    if collection.empty?
      return
    end

    if collection[0] % 2 != 0
      result << collection[0]
    end

    odd_collector(collection.slice(1, collection.length), result)
  end

  odd_collector(arr, result)
  result
end

p collect_odd_values([1, 2, 3, 4, 5, 6, 7, 8, 9])

File: recursion/collect-odd-values.js
function collectOddValues(arr){
  // Array of numbers
  // Odd values from array\
  let oddNumsArr = []

  let oddSelector =  (collection) => { // push odd numbers in my arr
    if(collection.length === 0) {
      return;
    }

    if(collection[0] % 2 !== 0) {
      oddNumsArr.push(collection[0])
    }

    oddSelector(collection.slice(1)) 
  }

  oddSelector(arr)

  console.log(oddNumsArr)
}

collectOddValues([1,2,3,4,5,6,7,8,9])